ang/civicase/Dashboard.js

```diff
@@ -0,0 +1,207 @@
+        sequential: 1,
+        is_current_revision: 1,
+        is_test: 0,
+        return: ['case_id', 'activity_type_id', 'subject', 'activity_date_time', 'status_id', 'target_contact_name', 'assignee_contact_name', 'is_overdue', 'is_star', 'file_id', 'case_id.case_type_id', 'case_id.status_id', 'case_id.contacts']
+      }, $scope.activityFilters);
+      // recent communication
+      apiCalls.push(['Activity', 'get', _.extend({
+        'activity_type_id.grouping': {LIKE: '%communication%'},
+        'status_id.filter': 1,
+        options: {limit: activitiesToShow, sort: 'activity_date_time DESC'}
+      }, params)]);
+      apiCalls.push(['Activity', 'getcount', _.extend({
+        'activity_type_id.grouping': {LIKE: '%communication%'},
+        'status_id.filter': 1,
+        is_current_revision: 1,
+        is_test: 0
+      }, $scope.activityFilters)]);
+      // next milestones
+      apiCalls.push(['Activity', 'get', _.extend({
+        'activity_type_id.grouping': {LIKE: '%milestone%'},
+        'status_id.filter': 0,
+        options: {limit: activitiesToShow, sort: 'activity_date_time ASC'}
+      }, params)]);
+      apiCalls.push(['Activity', 'getcount', _.extend({
+        'activity_type_id.grouping': {LIKE: '%milestone%'},
+        'status_id.filter': 0,
+        is_current_revision: 1,
+        is_test: 0
+      }, $scope.activityFilters)]);
+      crmApi(apiCalls).then(function (data) {
+        $scope.$broadcast('caseRefresh');
+        $scope.summaryData = data[apiCalls.length - 5].values;
+        $scope.dashboardActivities.recentCommunication = _.each(data[apiCalls.length - 4].values, formatActivity);
+        $scope.dashboardActivities.recentCommunicationCount = data[apiCalls.length - 3];
+        $scope.dashboardActivities.nextMilestones = _.each(data[apiCalls.length - 2].values, formatActivity);
+        $scope.dashboardActivities.nextMilestonesCount = data[apiCalls.length - 1];
+      });
+    };
+
+    /**
+     * Bind route paramaters to scope variables
+     */
+    function bindRouteParamsToScope () {
+      $scope.$bindToRoute({ param: 'dtab', expr: 'activeTab', format: 'int', default: 0 });
+      $scope.$bindToRoute({ param: 'dbd', expr: 'showBreakdown', format: 'bool', default: false });
+      $scope.$bindToRoute({ param: 'drel', expr: 'filters.caseRelationshipType', format: 'raw', default: 'is_involved' });
+    }
+
+    /**
+     * Watcher for caseRelationshipType
+     *
+     * @param {String} newValue
+     */
+    function caseRelationshipTypeWatcher (newValue) {
+      newValue === 'is_case_manager'
+        ? $scope.activityFilters.case_filter.case_manager = [CRM.config.user_contact_id]
+        : delete ($scope.activityFilters.case_filter.case_manager);
+
+      newValue === 'is_involved'
+        ? $scope.activityFilters.case_filter.contact_id = [CRM.config.user_contact_id]
+        : delete ($scope.activityFilters.case_filter.contact_id);
+    }
+
+    /**
+     * Prepate case filter options
```

Comment:
Typo here. Also `Prepare case filter option` is same as function name `prepareCaseFilterOption `. Can we have better legible description. 
Some thing like 
```
    /**
     * Transform filter options to an object with a pattern understood by ui-multi-select.
     */
 function prepareCaseFilterOption ()
```
PS: I am very bad at it TBH.

Reply:
Thanks for noticing.
Although we are not really transforming any object. But changed the doc to the following
`Prepare case filter options for crmUiSelect`

let me know if this is fine.